src/material-examples/tree-flat-overview/tree-flat-overview-example.ts

```diff
@@ -117,31 +53,27 @@ export class FileDatabase {
   selector: 'tree-flat-overview-example',
   templateUrl: 'tree-flat-overview-example.html',
   styleUrls: ['tree-flat-overview-example.css'],
-  providers: [FileDatabase]
 })
 export class TreeFlatOverviewExample {
-  treeControl: FlatTreeControl<FileFlatNode>;
-  treeFlattener: MatTreeFlattener<FileNode, FileFlatNode>;
-  dataSource: MatTreeFlatDataSource<FileNode, FileFlatNode>;
-
-  constructor(database: FileDatabase) {
-    this.treeFlattener = new MatTreeFlattener(this.transformer, this._getLevel,
-      this._isExpandable, this._getChildren);
-    this.treeControl = new FlatTreeControl<FileFlatNode>(this._getLevel, this._isExpandable);
-    this.dataSource = new MatTreeFlatDataSource(this.treeControl, this.treeFlattener);
-
-    database.dataChange.subscribe(data => this.dataSource.data = data);
+  private transformer = (node: FoodNode, level: number) => {
+    return {
+      expandable: !!node.children && node.children.length > 0,
+      name: node.name,
+      level: level,
+    };
   }
 
-  transformer = (node: FileNode, level: number) => {
-    return new FileFlatNode(!!node.children, node.filename, level, node.type);
-  }
+  treeControl = new FlatTreeControl<ExampleFlatNode>(
+      node => node.level, node => node.expandable);
 
-  private _getLevel = (node: FileFlatNode) => node.level;
+  treeFlattener = new MatTreeFlattener(
+      this.transformer, node => node.level, node => node.expandable, node => node.children);
 
-  private _isExpandable = (node: FileFlatNode) => node.expandable;
+  dataSource = new MatTreeFlatDataSource(this.treeControl, this.treeFlattener, TREE_DATA);
 
-  private _getChildren = (node: FileNode): Observable<FileNode[]> => observableOf(node.children);
+  constructor() {
+    this.dataSource.data = TREE_DATA;
```

Comment:
Same note as on the other example: it shouldn't be necessary to pass in the data here again. If it does make a difference, maybe it's a bug?